event_model/__init__.py

```diff
@@ -751,6 +751,7 @@ def compose_event_page(*, descriptor, event_counter, data, timestamps, seq_num,
 def compose_event(*, descriptor, event_counter, data, timestamps, seq_num=None,
                   filled=None, uid=None, time=None, validate=True):
     if seq_num is None:
+        event_counter[descriptor['name']] += 1
```

Comment:
This will increment the `seq_num` even if the call later errors out below. The original only increments if a document is produced. Can we fix this in `compose_stop` instead?